### DISCUSSION TITLE
What do you use SkyPilot for?

### DISCUSSION THREAD

**Discussion by romilbhardwaj** [skypilot-org/skypilot#1016] [General]
While SkyPilot supports a wide range of machine learning workloads (from [training Detectron2 models](https://github.com/skypilot-org/skypilot/blob/master/examples/detectron2_app.yaml) to [fine-tuning BERT for question answering](https://github.com/skypilot-org/skypilot/blob/master/examples/spot/bert_qa.yaml) on spot instances), we are curious to hear how you use SkyPilot! 

Please share your use case for SkyPilot here and any suggestions you may have on how SkyPilot can support it better. We'll be very happy to work with you on making your project a success on SkyPilot!

**Comment by romilbhardwaj** [comment]
I recently used SkyPilot's [`sky gpunode`](https://sky-proj-sky.readthedocs-hosted.com/en/latest/reference/interactive-nodes.html) functionality to get cloud instances for running some artifact evaluations for [Eurosys 2022](https://sysartifacts.github.io/eurosys2022/index). The ability to quickly spawn VMs with different GPU types (using the `--gpu` flag) made my job much easier!

**Comment by concretevitamin** [comment]
SkyPilot's **repeatable setup** is very helpful to me. By writing the setup once in a YAML, I can be confident in launching as many new VMs (and tearing them down whenever) as needed, knowing that my job will run the same across many clouds and across a long time scale!

This most recently helped me in the open sourcing of the [Balsa](https://github.com/balsa-project/balsa/) project, where I used SkyPilot to rerun many experiments on AWS and Azure to check that they can be reproduced by others.

**Comment by gilv** [comment]
@romilbhardwaj @concretevitamin any thoughts on using SkyPilot to optimize Spark jobs? (for example, SkyPilot will deploy the job on the cluster that is 'close' to the data, or using other optimization strategies to choose the right location to execute the job, etc.. )

**Comment by istoica** [comment]
Hi @gilv, this is a great question and we are looking into it. In fact, we are also looking at leveraging existing Spark hosted services like EMR, DataProc, HDFInsight, Databricks, etc. Let's keep in touch.

**Comment by asaiacai** [comment]
Love the tool! For me the killer feature was being able to easily run multinode torch DDP. In particular having the head node ip exposed in the run scripts made life a lot simpler. It makes paid options like MosaicML feel moot. I didn't like having to navigate the cloud consoles for different providers so having a unified interface for submitting jobs was good. It was really nice having sky serve added and also being able to ssh and access the logs easily from the CLI of the different workers.

I'm interested in seeing how Skypilot DAGs and k8s plays out. I'm curious what you guys see is the future of the project beyond what was in the original SkyPilot paper.

**Reply by concretevitamin** [reply]
Glad to see the feedback @asaiacai! Stay tuned on the new support for serving workloads and k8s clusters running in the Sky.
